zerver/webhooks/gocd/view.py

```diff
@@ -26,31 +37,75 @@ def api_gocd_webhook(
-
-    body = MESSAGE_TEMPLATE.format(
-        modifications["user_name"].tame(check_string),
-        result,
-        emoji,
-        build_link,
-        modifications["comment"].tame(check_string),
+    type = payload["type"].tame(check_string)
+    if type == "stage":
+        body = get_body(payload)
+        topic_name = get_topic(payload)
+        check_send_webhook_message(request, user_profile, topic_name, body)
+    return json_success(request)
+
+
+def get_topic(payload: WildValue) -> str:
+    return TOPIC_TEMPLATE.format(
+        pipeline=payload["data"]["pipeline"]["name"].tame(check_string),
+        stage=payload["data"]["pipeline"]["stage"]["name"].tame(check_string),
     )
-    branch = material["description"].tame(check_string).split(",")
-    topic_name = branch[0].split(" ")[1]
 
-    check_send_webhook_message(request, user_profile, topic_name, body)
 
-    return json_success(request)
+def get_commit_details(payload: WildValue) -> str:
+    build_cause = payload["data"]["pipeline"]["build-cause"]
+    material = build_cause[0]["material"]
+    url_base = material["git-configuration"]["url"].tame(check_string)
+    revision = build_cause[0]["modifications"][0]["revision"].tame(check_string)
+    commit_sha = get_short_sha(revision)
+    url = f"{url_base}/commit/{commit_sha}"
+    branch = material["git-configuration"]["branch"].tame(check_string)
+    return COMMIT_INFO_TEMPLATE.format(
+        commit_details=commit_sha,
+        commit_link=url,
+        branch_name=branch,
+    )
+
+
+def get_jobs_details(pipeline_data: WildValue) -> str:
+    job_dict_list = pipeline_data["stage"]["jobs"]
+    formatted_job_dict = defaultdict(list)
+    job_details_template = ""
+
+    for job in job_dict_list:
+        job_name = job["name"].tame(check_string)
+        job_result = job["result"].tame(check_string)
+        formatted_job_dict[job_result].append(f"`{job_name}`")
+
+    for key in formatted_job_dict:
+        formatted_job_list = ", ".join(formatted_job_dict[key])
+        job_details_template += f"- **{key}**: {formatted_job_list}\n"
+
+    return job_details_template
+
+
+def get_body(payload: WildValue) -> str:
+    pipeline_data = payload["data"]["pipeline"]
+    body_details = {
+        "commit_details": get_commit_details(payload),
+        "status": pipeline_data["stage"]["state"].tame(check_string).lower(),
+        "pipeline": pipeline_data["name"].tame(check_string),
+        "stage": pipeline_data["stage"]["name"].tame(check_string),
+        "start_time": pipeline_data["stage"]["create-time"].tame(check_string),
+    }
+
+    if body_details["status"] == "building":
+        return SCHEDULED_BODY_TEMPLATE.format(**body_details)
+
+    result = pipeline_data["stage"]["result"].tame(check_string)
+    body_details.update(
+        {
+            "result": result,
+            "emoji": ":check:" if result == "Passed" else ":warning:",
```

Comment:
Based on [this CZO discussion](https://chat.zulip.org/#narrow/channel/137-feedback/topic/markdown.20alerts.20syntax/near/1961518), should we maybe use `":green_circle:"` and `":red_circle:"` here? Changing this would mean updating the generated screenshot. This isn't an alert/monitoring integration, but it might be good to align with that general direction for now.